app/services/find/course_type_answer_determiner.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Find
+  class CourseTypeAnswerDeterminer
+    attr_reader :age_group, :visa_status, :university_degree_status
+
+    def initialize(age_group:, visa_status:, university_degree_status:)
+      @age_group = age_group
+      @visa_status = visa_status
+      @university_degree_status = university_degree_status
+    end
+
+    def show_undergraduate_courses?
+      !further_education? && university_degree_status.present? && no_degrees? && does_not_require_visa_sponsorship?
+    end
+
+    def not_elibible_for_undergraduate_courses?
+      !further_education? && university_degree_status.present? && no_degrees? && require_visa_sponsorship?
+    end
```

Comment:
Is `no_degrees?` correct -- doesn't the fact that someone *doesn't* have a degree mean they are more likely to be interested in undergraduate courses? I would think someone is not eligible for undergraduate courses if they already have a degree.

Reply:
To the context of only one conditional yes. But the exist page is show if the candidate answer no degrees and that requires visa sponsorship.

Reply:
`show_exit_page?` is definitely better. Thank you Lori. And @avinhurry on  https://github.com/DFE-Digital/publish-teacher-training/pull/4428#discussion_r1710322601